sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/change_feed_utils.py

```diff
@@ -0,0 +1,148 @@
+
+    if len(args) > 0:
+        keys = [
+            'partition_key_range_id',
+            'is_start_from_beginning',
+            'continuation',
+            'max_item_count',
+        ]
+        for i, value in enumerate(args):
+            key = keys[i]
+
+            if key in kwargs:
+                raise TypeError(f"'query_items_change_feed()' got multiple values for argument '{key}'.")
+
+            kwargs[key] = value
+
+def validate_kwargs(
+        kwargs: Dict[str, Any]
+    ) -> None:
+    """Validate keyword arguments(kwargs).
+    The values of keyword arguments must match the expect type and conditions. If the conditions do not match,
+    errors will be raised with the error messages and possible ways to correct the errors.
+
+    :param kwargs: Keyword arguments to verify for query_items_change_feed API
+    :keyword mode: Must be one of the values in the Enum, 'ChangeFeedMode'.
+        If the value is 'ALL_VERSIONS_AND_DELETES', the following keywords must be in the right condition:
+            - 'partition_key_range_id': Cannot be used at any time
+            - 'is_start_from_beginning': Must be 'False'
+            - 'start_time': Must be "Now"
+    :paramtype mode: Optional[Literal["LatestVersion", "AllVersionsAndDeletes"]]
+    :keyword partition_key_range_id: Deprecated Warning.
+    :paramtype partition_key_range_id: str
+    :keyword is_start_from_beginning: Deprecated Warning. Cannot be used with 'start_time'.
+    :paramtype is_start_from_beginning: bool
+    :keyword start_time: Must be in supported types.
+    :paramtype start_time: Union[~datetime.datetime, Literal["Now", "Beginning"]]
+    :type kwargs: dict[str, Any]
+    """
+    # Filter items with value None
+    kwargs = {key: value for key, value in kwargs.items() if value is not None}
+
+    # Validate the keyword arguments
+    if "mode" in kwargs:
+        mode = kwargs["mode"]
+        if mode not in CHANGE_FEED_MODES:
+            raise ValueError(
+                f"Invalid mode was used: '{kwargs['mode']}'."
+                f" Supported modes are {CHANGE_FEED_MODES}.")
+
+        if mode == 'AllVersionsAndDeletes':
+            if "partition_key_range_id" in kwargs:
+                raise ValueError(
+                    "'AllVersionsAndDeletes' mode is not supported if 'partition_key_range_id'"
+                    " was used. Please use 'feed_range' instead.")
+            if "is_start_from_beginning" in kwargs and kwargs["is_start_from_beginning"] is not False:
+                raise ValueError(
+                    "'AllVersionsAndDeletes' mode is only supported if 'is_start_from_beginning'"
+                    " is 'False'. Please use 'is_start_from_beginning=False' or 'continuation' instead.")
+            if "start_time" in kwargs and kwargs["start_time"] != "Now":
+                raise ValueError(
+                    "'AllVersionsAndDeletes' mode is only supports if 'start_time' is 'Now'."
```

Comment:
I would still make these checks case-insensitive.
```suggestion
                    "'AllVersionsAndDeletes' mode is only supported if 'start_time' is 'Now'."
```

Reply:
Done